### PR TITLE
[Feat] Added Opik integration for logging and evaluation

### PR DESCRIPTION
**Pull request by dsblank**
## Title

Added Comet's Opik open-source observability integration for logging and evaluation. 

For more information on Opik, see: https://github.com/comet-ml/opik

Hope you find this useful!

## Type

🆕 New Feature

## Changes

1. Adds documentation: docs/my-website/docs/observability/opik_integration.md
2. Lists Opik in sidebar
3. Adds the integration: litellm/integrations/opik.py
4. Adds tests: litellm/tests/test_opik.py

### UI Changes

Screenshot from 2024-09-13 08-36-01

## Testing

All Tests Pass:

```
# Set api keys in environment
pytest litellm/tests/test_opik.py
```

Screenshot from 2024-09-13 08-18-16

Tests can run with openai and anthropic keys. Traces and spans are logged to a local or hosted Opik server.

### CODE REVIEW

**Comment by ghost** [comment]
Hi @dsblank is Opik otel compatible? 

If so, we already have an otel logger - could you just inherit from / use that? (it'll mean fewer dependencies on litellm docker images) 

https://github.com/BerriAI/litellm/blob/cd8d7ca9156a5fc2510db1ef0d43956d3239eccf/litellm/integrations/opentelemetry.py#L41

See the arize_ai integration for reference - https://github.com/BerriAI/litellm/blob/cd8d7ca9156a5fc2510db1ef0d43956d3239eccf/litellm/integrations/arize_ai.py#L17

**Comment by dsblank** [comment]
@krrishdholakia, I'm checking to be 100% sure, but I don't think so.

**Comment by dsblank** [comment]
@krrishdholakia , no sorry: opik is not OTEL compatible.

Let me know if there is anything else you need.

**Comment by dsblank** [comment]
Just checking if you need any further information.

BTW, we had the official Opik launch! https://www.comet.com/site/products/opik/

File: litellm/litellm_core_utils/litellm_logging.py [hunk 1]
```diff
@@ -121,6 +123,7 @@
 berrispendLogger = None
 supabaseClient = None
 liteDebuggerClient = None
+opikLogger = None
```
Comment (ishaan-jaff) [inline litellm/litellm_core_utils/litellm_logging.py]: we're moving away from adding loggers like this to litellm_logging.py - please make use of the CustomLogger() class. See `langsmith.py` on how we implement custom loggers

File: litellm/litellm_core_utils/litellm_logging.py [hunk 2]
```diff
@@ -1263,6 +1266,15 @@ def success_handler(
                             print_verbose=print_verbose,
                             callback_func=callback,
                         )
+                    if callback == "opik":
```
Comment (ishaan-jaff) [inline litellm/litellm_core_utils/litellm_logging.py]: there should be no code in `litellm_logging.py`

File: litellm/integrations/opik.py [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    else:
+        current_span_data = get_current_span_data()
+
+    if current_span_data:
+        return current_span_data.id
+    else:
+        return None
+
+
+def get_current_trace_id(metadata: Dict[str, Any]) -> str:
```
Comment (ishaan-jaff) [inline litellm/integrations/opik.py]: these are utils functions. Please make a folder called optik/ 

use this structure:

optik/utils.py -> place optik utils here 
optik/optik.py -> place main logging code here

File: litellm/integrations/opik.py [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+            value = redact_secrets(value)
+            if key == "api_key":
+                value = "***REDACTED***"
+            redacted_dict[key] = value
+        return redacted_dict
+    else:
+        return item
+
+
+class OpikLogger():
```
Comment (ishaan-jaff) [inline litellm/integrations/opik.py]: should inherit from litellm.CustomLogger()

File: litellm/integrations/opik.py [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+    Opik Logger for logging events to an Opik Server
+    """
+    def __init__(self):
+        try:
+            import opik
+            self.opik = opik
+        except ImportError:
+            self.opik = None
+
+    def log_event(
```
Comment (ishaan-jaff) [inline litellm/integrations/opik.py]: you just need to implement async_log_success_event and log_success_event. See langsmith.py or datadog.py as a reference @dsblank

File: litellm/integrations/opik.py [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+        current_span_data = get_current_span_data()
+
+    if current_span_data:
+        return current_span_data.id
+    else:
+        return None
+
+
+def get_current_trace_id(metadata: Dict[str, Any]) -> str:
+    from opik.opik_context import get_current_trace_data
```
Comment (ishaan-jaff) [inline litellm/integrations/opik.py]: can you implement this without the `optik` python package. Our users complain integrations don't work when they require additional dependencies. 

You can see how we do httpx logging in langsmith or datadog.py

**Review by ishaan-jaff — CHANGES_REQUESTED** [review]
requested changes @dsblank

**Comment by dsblank** [comment]
Thanks for the information on your changing approach! I'll take a look.

**Comment by jverre** [comment]
@krrishdholakia This ended up being a branch with no changes, I'm currently working on re-writting the integration so that it doesn't depend on the Opik SDK and use the new approach to using loggers in LiteLLM. It will be ready later today for your review

**Comment by ishaan-jaff** [comment]
can you open a new PR with the feedback we gave @jverre

**Pull request merged**